### PR TITLE
Fix findbugs

### PR DESCRIPTION
**Pull request by AndyScherzinger**
* replace int based for-loops
* literal string comparisons passing the literal as an argument
* fix trims a String temporarily
* use isEmpty() to check for empty collection instead of size <= 0
* remove unused parameters
* remove locally saving values right before returning them

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
pinging @ardevd @tobiasKaminsky @mario for a code review :) :heart:

**Comment by mario** [comment]
👍

File: src/main/java/com/owncloud/android/utils/ReceiversHelper.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public static void registerPowerChangeReceiver() {
         BroadcastReceiver broadcastReceiver = new BroadcastReceiver() {
             @Override
             public void onReceive(Context context, Intent intent) {
-                if (intent.getAction() != null && intent.getAction().equals(Intent.ACTION_POWER_CONNECTED)) {
+                if (intent.getAction() != null && Intent.ACTION_POWER_CONNECTED.equals(intent.getAction())) {
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/utils/ReceiversHelper.java]: first check can be ommitted, or?

**Comment by tobiasKaminsky** [comment]
Very nice :tada:

File: src/main/java/com/owncloud/android/utils/ReceiversHelper.java (hunk 1 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/utils/ReceiversHelper.java]: right, fixed via https://github.com/nextcloud/android/pull/2860/commits/d736519c9862f30ceb7f4f938f3ed598de638d2f waiting for the checks to turn green :)

**Pull request merged**